.gitignore

```diff
@@ -1,3 +1,2 @@
 .idea
-composer.lock
```

Comment:
> It makes maintaining a project with PRs and different branches hard because you'll get constant merge conflicts with the lock file.

I, personally, don't put much weight in the merge conflicts with the composer file with a library, this point is far more relevant in complete applications where the composer.json changes more regularly. If all is well a library's composer.json only changes seldomly.

> Surely the point of tests is to check it currently works rather than worked with some old dependencies. You want to know if something broke straight away.

This point is a bit contested; on one hand I agree with you but on the other hand should you, technically, be able to test with every version of a supported dependency.

To be frank: it can be considered poor for support to state in your composer.json that you accept a dependency between 1.0 and 2.0. Technically you'd have to test every released version to see if it doesn't break.

Once you accept that you can't support all versions of dependency you will need to take a point that you can support, as library author.

Here you have two choices: the latest or a version that you know was stable.

From a support perspective I prefer the latter, I don't want to know when a dependency breaks my build; I want to know when _I_ break the build.